trade-finance-manager-api/src/v1/helpers/amendment.helpers.js

```diff
@@ -499,25 +499,33 @@ const calculateAmendmentDateTenor = async (coverEndDate, existingFacility) => {
 };
 
 // populates tfmObject with date values
-const addLatestAmendmentDates = async (tfmObject, latestDate, facilityId) => {
-  // if there is a latest coverEndDate
-  if (latestDate?.coverEndDate) {
-    const { coverEndDate } = latestDate;
-    const existingFacility = await api.findOneFacility(facilityId);
+const addLatestAmendmentDates = async (tfmObject, latestCoverEndDateResponse, latestFacilityEndDateDataResponse, facilityId) => {
```

Comment:
I'm new to adding docstrings - I can't work out what I can add here other than: 
```
/**
 * Populates the tfmObject with date values
 * @param tfmObject
 * @param latestCoverEndDateResponse
 * @param latestFacilityEndDateDataResponse
 * @param facilityId
 * @returns {Object}
 */
```
Given it returns an object that may or may not already contain `value` and `exposure` (depending on if this was passed in with the original `tfmObject`) and may or may not contain `coverEndDate` and  ` amendmentExposurePeriodInMonths` and may or may not contain `facilityEndDate`). 

But I might just not quite understand how to express types in these docstrings! So do let me know if there's a way to do this well :)

Reply:
So I think that `tfmObject` has the type `TfmFacility`, but I'm not sure what `latestCoverEndDateResponse` and `latestFacilityEndDateDataResponse` are. However, based on how they are used, I think this could be something like
```typescript
/**
 * Populates the tfmObject with date values
 * @param {import('@ukef/dtfs2-common').TfmFacility} tfmObject
 * @param {{ coverEndDate?: string } | undefined} latestCoverEndDateResponse
 * @param {{ facilityEndDate?: string } | undefined} latestFacilityEndDateDataResponse
 * @param {string} facilityId
 * @returns {import('@ukef/dtfs2-common').TfmFacility & { coverEndDate?: string, facilityEndDate?: string }}
 */
```

This is based on guesses about the type from the implementation of the function